src/nebula_core/nebula_core_hw_interfaces/include/nebula_core_hw_interfaces/nebula_hw_interfaces_common/connections/udp.hpp

```diff
@@ -365,6 +295,12 @@ class UdpSocket
     return *this;
   }
 
+  void close()
```

Comment:
After closing, there is no way to re-open the socket, except for creating a new one.

I would suggest removing this function again, and destroying the UdpSocket instance instead at the call site.